File: foosball/ClientApp/src/team.ts
import { TeamMember } from "./team-member";

export class Team {
    constructor(name: string, members: TeamMember[] = []){
        this.teamId = 0;
        this.name = name;
        this.score = 0;
        this.members = members;
    }

    public teamId: number;
    public name: string;
    public score: number;
    public members: Array<TeamMember>;
}

File: foosball/Database/Models/Team.cs
namespace foosball.Database.Models {
    public class Team {
        public int TeamId { get; set; }
        public string Name { get; set; }
        public int Score { get; set; }
        public virtual ICollection<TeamMember> Members { get; set; }
        public Team(ICollection<TeamMember> members) {
            Members = members;
        }

        public Team() { }
    }
}
